gsy_framework/sim_results/kpi.py

```diff
@@ -67,29 +71,29 @@ def _accumulate_total_energy_demanded(self, area_dict: Dict, core_stats: Dict):
     def _accumulate_self_production(self, trade: Dict):
```

Comment:
In `_accumulate_self_consumption` we're checking the seller origin id and buyer origin id, but we're only checking the seller origin id here. Is there any reason?

Reply:
These 2 methods update different members and have a different purpose. The `_accumulate_self_production` updates the total_energy_produced_Wh (which tracks the energy produced in all descendants of the area that we calculate these metrics for), while the `_accumulate_self_consumption` updates the total_self_consumption_Wh (which tracks the self-consumed energy for all descendants - energy that has been produced AND consumed by descendants).

Reply:
Your comment made it clear to me, but I doubt if the `_accumulate_self_consumption` is a proper name for a function which tracks both production and consumption. Please correct me if I'm wrong.